.devcontainer/devcontainer.json

```diff
@@ -1,22 +1,22 @@
-// For format details, see https://aka.ms/devcontainer.json. For config options, see the
-// README at: https://github.com/devcontainers/templates/tree/main/src/python
 {
-  "name": "Python 3",
-  // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
-  "image": "mcr.microsoft.com/devcontainers/python:1-3.11-bullseye"
-
-  // Features to add to the dev container. More info: https://containers.dev/features.
-  // "features": {},
-
-  // Use 'forwardPorts' to make a list of ports inside the container available locally.
-  // "forwardPorts": [],
-
-  // Use 'postCreateCommand' to run commands after the container is created.
-  // "postCreateCommand": "pip3 install --user -r requirements.txt",
-
-  // Configure tool-specific properties.
-  // "customizations": {},
-
-  // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
-  // "remoteUser": "root"
-}
+    "name": "Wagtail AI",
+    "image": "mcr.microsoft.com/devcontainers/python:1-3.11-bullseye",
+    "postCreateCommand": "pip install tox && tox devenv -e interactive && npm ci && ./venv/bin/pre-commit install",
```

Comment:
**Question:** What does this do? Does this install all the dependencies to the container? I guess we don't actually end up with tox installed in the environment, is that expected?

Reply:
It installs `tox` globally, then uses it to generate a virtual environment with all the depdendencies installed, then switches to the venv.

At that point, `tox` is still available globally and the application dependencies are all available in the virtual env.